acceptance/tests/pxp-module-puppet/run_puppet_agent_non_root.rb

```diff
@@ -0,0 +1,59 @@
+require 'pxp-agent/test_helper.rb'
+require 'yaml'
+
+test_name  'Run puppet agent as non-root' do
+
+  platform = agents.first.platform
+  skip_test "Test is not compatible with #{platform}" if platform =~ /windows/
+
+  step 'create non-root user on all nodes' do
+    on(agents, puppet("resource user foo ensure=present managehome=true"))
+
+    #TODO fix this only one agent ???
```

Comment:
Plan to address this?

Reply:
It seems to be a pattern that we use in all the other tests, however I didn't see anywhere that we specify more than one agent in a test configuration.

Reply:
We don’t. It was broken for awhile and no one noticed.

Reply:
Sorry @MikaelSmith Im not sure I understand, are you saying that we don't ever use more than one agent?

I can make the test loop through all the agents if you feel its valuable, but if not we can just address the target with the helper method `agent`

Reply:
I was saying we don't ever test with more than one agent in practice. It's supported by Beaker and the test suite, but never exercised outside people doing it from their laptops.

Reply:
In that case I'll put the loop in.